src/components/MessagesList/MessagesGroup/Message/MessagePart/MessageBody.vue

```diff
@@ -420,6 +423,44 @@ export default {
 				}
 			}
 		},
+
+		async handleInteraction(id) {
+			const parentId = id.split('-markdown-input-')[0]
+			const index = Array.from(document.querySelectorAll(`span[id^="${parentId}-markdown-input-"]`)).findIndex((el) => el.id.includes(id))
+			if (index === -1) {
+				return
+			}
+			let checkBoxIndex = 0
+			const lines = this.message.split('\n')
+			for (let i = 0; i < lines.length; i++) {
+				if (lines[i].includes('[ ]') || lines[i].includes('[x]')) {
+					if (checkBoxIndex === index) {
+						const isChecked = lines[i].includes('[x]')
+						if (isChecked) {
+							lines[i] = lines[i].replace('[x]', '[ ]')
```

Comment:
does this replace only the first occurrence?

- [ ] Yes - [ ]
- [ ] No - [ ]

Reply:
Yes, and markdown doesn't read boxes that have other characters than space before
![image](https://github.com/nextcloud/spreed/assets/84044328/7caa3a2e-e193-4203-8614-c66a96644138)

Reply:
> Should be fine, although I'd use - [ ] with a dash

no because it can also be  `-    [ ]` ( up to 4 spaces). I adjusted the criteria for the line to be considered as a checkbox markdown.

Reply:
couldn't be 6? but I'm fine with merging it as-is

Reply:
Oh, and also should message should be editable :warning: 
Please refer to the https://github.com/nextcloud/spreed/issues/11402#issuecomment-1997493072